### PR TITLE
Deploy composed Storybook previews from CI

### PR DESCRIPTION
**Pull request by noe-charmet**
## Summary

Move composed Storybook preview publication to GitHub Actions and Cloudflare Pages Direct Upload so the provider does not build the repository or consume a hosted deployment queue.

The workflow now:

- Uses Turbo affected planning and forced paths to select applications.
- Builds selected applications with repository-owned, environment-specific inputs.
- Assembles and validates the composed Storybook artifact in CI.
- Uploads prebuilt output to the configured Cloudflare Pages project or projects.
- Verifies the exact commit metadata and configured endpoints before reporting success.
- Publishes readable per-application GitHub Deployments and a job summary.
- Cancels superseded pull-request runs and protects Cloudflare credentials from fork code.

The reusable `@shipfox/cloudflare-pages` package provides planning, build, deploy, verification, GitHub lifecycle, and summary commands for other repositories to adopt the same flow.

Production uploads use an explicit `main` branch and validate Wrangler's structured production metadata. The workflow also handles fork base-fetch failures and subprocess timeout, pipe, and cancellation edge cases.

Validation completed with `mise exec -- pnpm turbo run check type type:emit test build --filter=@shipfox/cloudflare-pages...` (28/28 tasks, 26 Cloudflare Pages tests), workflow YAML parsing, README readability review, and `git diff --check`.

Vercel project archival and external cut-over remain tracked separately in ENG-1180.

Closes ENG-1177

### CODE REVIEW

File: tools/cloudflare-pages/src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+  process.stdout.write(
+    `${resultWithDuration.ok ? 'Verified' : 'Verification failed'} ${resultWithDuration.apps.length} Cloudflare Pages app(s)\n`,
+  );
+  if (!resultWithDuration.ok) throw new Error(resultWithDuration.errors.join('; '));
+}
+
+async function runGitHub(action: string, options: CliOptions): Promise<void> {
+  if (action === 'assert-current') {
+    const result = await assertCurrentCommit({
+      repository: option(options, 'repository'),
```
[the remark this answers is not in the document]
Comment (noe-charmet) [reply tools/cloudflare-pages/src/cli.ts:264]: Addressed in f48c878db. createGitHubDeployments now defaults the omitted repository option to GITHUB_REPOSITORY, which is how the workflow invocation supplies the repository. The regression test “uses the GitHub repository from the CI environment for deployment registration” covers this path.
